functions/collection.test.ts

```diff
@@ -0,0 +1,77 @@
+const collections = [
+  {
+    address: '0xed5af388653567af2f388e6224dc7c4b3241c544',
+    collectionName: 'Azuki',
+    image:
+      'https://i.seadn.io/gae/H8jOCJuQokNqGBpkBN5wk1oZwO7LM8bNnrHCaekV2nKjnCqw6UB5oaH8XyNeBDj6bA_n1mjejzhFQUP3O1NfjFLHr3FOaeHcTOOT?w=500&auto=format',
+  },
+  {
+    address: '0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d',
+    collectionName: 'Bored Ape Yacht Club',
+    image:
+      'https://i.seadn.io/gae/Ju9CkWtV-1Okvf45wo8UctR-M9He2PjILP0oOvxE89AyiPPGtrR3gysu1Zgy0hjd2xKIgjJJtWIc0ybj4Vd7wv8t3pxDGHoJBzDB?w=500&auto=format',
+  },
+  {
+    address: '0x49cf6f5d44e70224e2e23fdcdd2c053f30ada28b',
+    collectionName: 'CLONE X - X TAKASHI MURAKAMI',
+    image:
+      'https://i.seadn.io/gae/XN0XuD8Uh3jyRWNtPTFeXJg_ht8m5ofDx6aHklOiy4amhFuWUa0JaR6It49AH8tlnYS386Q0TW_-Lmedn0UET_ko1a3CbJGeu5iHMg?w=500&auto=format',
+  },
+]
+
+test('should inject metadata for valid nfts', async () => {
+  for (const nft of collections) {
+    const url = 'http://127.0.0.1:3000/nfts/collection/' + nft.address
+    const req = new Request(url)
+    const res = await fetch(req)
+    const body = await res.text()
+    expect(body).toMatchSnapshot()
+    expect(body).toContain(nft.collectionName + ' on Uniswap')
+    expect(body).toContain(nft.image)
+    expect(body).toContain(url)
+    expect(body).toContain(`<meta property="og:title" content = "${nft.collectionName} on Uniswap"/>`)
+    expect(body).toContain(`<meta property="og:image" content = "${nft.image}"/>`)
+    expect(body).toContain(`<meta property="og:image:width" content = "1200"/>`)
+    expect(body).toContain(`<meta property="og:image:height" content = "630"/>`)
+    expect(body).toContain(`<meta property="og:type" content = "website"/>`)
+    expect(body).toContain(`<meta property="og:url" content = "${url}"/>`)
+    expect(body).toContain(
+      `<meta property="og:image:alt" content = "https://app.uniswap.org/images/512x512_App_Icon.png"/>`
+    )
+    expect(body).toContain(`<meta property="twitter:card" content = "summary_large_image"/>`)
+    expect(body).toContain(`<meta property="twitter:title" content = "${nft.collectionName} on Uniswap"/>`)
+    expect(body).toContain(`<meta property="twitter:image" content = "${nft.image}"/>`)
+    expect(body).toContain(
+      `<meta property="twitter:image:alt" content = "https://app.uniswap.org/images/512x512_App_Icon.png"/>`
+    )
+  }
+}, 50000)
```

Comment:
Yeah, so on the initial run, yarn takes extra time to compile and setup the proxy, so the 50 second timeout is needed otherwise the tests will just fail as they wait for the server to get started